net.go

```diff
@@ -425,6 +426,122 @@ func (e *EurekaConnection) ScheduleSecureVIPAddressUpdates(addr string, await bo
 	return e.scheduleVIPAddressUpdates(addr, await, done, options), nil
 }
 
+// An InstanceSetSource holds a periodically updated set of instances registered with Eureka.
+type InstanceSetSource struct {
+	m         *sync.RWMutex
+	instances []*Instance
+	done      chan<- struct{}
+}
+
+func newInstanceSetSourceFromChannel(await bool, updates <-chan InstanceSetUpdate, done chan<- struct{}) *InstanceSetSource {
```

Comment:
I collapsed the function pairs, adding a "secure" `bool` parameter, but haven't uploaded the patch yet, because I'd like to hear from you first about exporting the `vipAddressKind` type (presumably as `VIPAddressKind`) and using that instead.